projects/dsp-ui/src/lib/core/session.service.ts

```diff
@@ -0,0 +1,215 @@
+     */
+    setSession(jwt: string, identifier: string, identifierType: 'email' | 'username') {
+
+        let session: Session;
+
+        if (jwt) {
+            this.updateDspApiConnection(jwt);
+        }
+
+        // get user information
+        this.dspApiConnection.admin.usersEndpoint.getUser(identifierType, identifier).subscribe(
+            (response: ApiResponseData<UserResponse>) => {
+                let sysAdmin = false;
+                const projectAdmin: string[] = [];
+
+                // get permission information: a) is user sysadmin? b) get list of project iri's where user is project admin
+                const groupsPerProjectKeys: string[] = Object.keys(response.body.user.permissions.groupsPerProject);
+
+                for (const key of groupsPerProjectKeys) {
+                    if (key === Constants.SystemProjectIRI) {
+                        sysAdmin = response.body.user.permissions.groupsPerProject[key].indexOf(Constants.SystemAdminGroupIRI) > -1;
+                    }
+
+                    if (response.body.user.permissions.groupsPerProject[key].indexOf(Constants.ProjectAdminGroupIRI) > -1) {
+                        projectAdmin.push(key);
+                    }
+                }
+
+                // store session information in browser's localstorage
+                // TODO: jwt will be removed, when we have a better cookie solution (DSP-261)
+                session = {
+                    id: this.setTimestamp(),
+                    user: {
+                        name: response.body.user.username,
+                        jwt: jwt,
+                        lang: response.body.user.lang,
+                        sysAdmin: sysAdmin,
+                        projectAdmin: projectAdmin
+                    }
+                };
+
+                // update localStorage
+                localStorage.setItem('session', JSON.stringify(session));
+            },
+            (error: ApiResponseError) => {
+                localStorage.removeItem('session');
+                console.error(error);
+            }
+        );
+    }
+
+    /**
+     * Validate intern session and check knora api credentials if necessary.
+     * If a json web token exists, it doesn't mean that the knora api credentials are still valid.
+     *
+     * @returns boolean
+     */
+    isSessionValid(): boolean {
+        // mix of checks with session.validation and this.authenticate
+        const session = JSON.parse(localStorage.getItem('session'));
+
+        const tsNow: number = this.setTimestamp();
+
+        if (session) {
+
+            this.updateDspApiConnection(session.user.jwt);
+
+            // check if the session is still valid:
+            if (session.id + this.MAX_SESSION_TIME <= tsNow) {
+                // the internal (dsp-ui) session has expired
+                // check if the api credentials are still valid
+
+                this.dspApiConnection.v2.auth.checkCredentials().subscribe(
```

Comment:
rather than subscribing, perform the necessary actions with `pipe`. The caller of this method should subscribe.